### ISSUE TITLE
Invisible characters in series name

### ISSUE THREAD

**Issue by langecode** [VictoriaMetrics/VictoriaMetrics#155]
**Describe the bug**
I am seeing a bit strange behaviour from some time series. It seems there is an "invisible" character that do not let me select the time series correctly.

Basically I am just trying to query information on the timeseries. If I do this with a regexp allowing any character at the end I get the series:
```
curl http://xx/select/2/prometheus/api/v1/series --data-urlencode 'match[]={__name__=~"GenBearTemp.*"}' --data-urlencode 'start=0'
{
  "status":"success",
  "data":[
    {"__name__":"GenBearTemp","db":"TenMinute","Park":"1","TurbineType":"V112"},
    {"__name__":"GenBearTemp","db":"TenMinute","Park":"2","TurbineType":"V112"},
    {"__name__":"GenBearTemp","db":"TenMinute","Park":"3","TurbineType":"V112"},
    {"__name__":"GenBearTemp","db":"TenMinute","Park":"4","TurbineType":"V112"}
  ]
}%
```

If I select the series specifically I get nothing:
```
curl http://xx/select/2/prometheus/api/v1/series --data-urlencode 'match[]={__name__="GenBearTemp"}' --data-urlencode 'start=0'
{"status":"success","data":[]}%
```

Data is ingested using the InfluxDB API. I am not sure whether we could have some "hidden" character at the end of the series name?

**To Reproduce**
Actually I have not yet found a distinct pattern - seems some names I can select specifically but some not.

**Expected behavior**
Would have expected the same output from the two queries.

**Version**
`1.25.2`

**Comment by valyala** [comment]
Let's try narrowing down the issue, since it cannot be reproduced at the moment.

1. How many `vmselect` nodes are in your cluster? Make sure all the `vmselect` nodes have identical set of `-storageNode` flags and these flags point to all the available `vmstorage` nodes in the cluster.

2. Try the following query:
```
curl http://xx/select/2/prometheus/api/v1/series --data-urlencode 'match[]=GenBearTemp' --data-urlencode 'start=0'
```
This query is equivalent to `match[]={__name__="GenBearTemp"}`, so it must return the same results.

3. Try exporting the data with `{__name__=~"metric.*"}` and `metric` queries via [http://xx/select/2/api/v1/export](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-export-time-series) and verifying whether they differ?

4. Could you provide additional `metric` examples, where `{__name__=~"metric.*"}` returns results for the given `metric`, while `{__name__="metric"}` and `metric` returns nothing?

5. Could you provide `metric` examples, where VictoriaMetrics always returns correct results for any query type mentioned above?

6. Could you provide the output of `http://xx/select/2/api/v1/label/__name__/values`?

7. Who ingests the data into the cluster and over which protocol?

**Comment by langecode** [comment]
Thank you very much for your suggestions- and for a very nice product. I will test this out. Just to answer the ones I can from the top of my head.

1.  We are pretty much just testing out VictoriaMetrics as of now - but our data estimates has us going towards the clustered version for scalability. We are currently running two select nodes but could easily try to scale that down to one to remove that variable.

3. Trying to use the export API I actually did see the time series that did not show up with the exact matching. However both the `series` API and the `query` and `query_range` APIs seems to have trouble with the exact matching. Did not try to do a matching with the export API though.

4. I did have several series showing the odd behaviour. No apparent pattern as to when one or the other was happening.

5. I did have series where an exact matching worked fine - but could not find any pattern as to why those names should act different. I will find some examples.

6. Will do

7. We ingest the data ourselves using the Influx API since it is IoT data it may be delayed and we use the Influx API with timestamps to ingest data. During the tests we have been ingesting historic data.

I will get back to you with some concrete examples. Thanks again for the work.
